src/feature/RansacFeatureSetMatcher.cpp

```diff
@@ -187,8 +190,10 @@ double RansacFeatureSetMatcher::matchSets(const std::vector<InterestPoint *> &re
 	    
 	    // Adapt the number of iterations
 	    if (m_adaptive){
-		double inlierProbability = double(correspondences.size())/double(data.size());
-		iterations = ceil(log(1. - m_successProbability)/log(1. - inlierProbability * inlierProbability));
+               inlierProbability = static_cast<double>(correspondences.size())/data.size();
+               const double inlierProbabilitySquared = inlierProbability * inlierProbability;
+               iterations = inlierProbabilitySquared > m_successProbability ? 1 :
```

Comment:
Optional: if you want put a comment saying that `m_successProbability < 1.0`, so we don't have to check for `log(0.0)`. You could also enforce `m_successProbability` to in the (0, 1) open interval.